src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,86 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_LENGTH = 1 << 4;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_LENGTH];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int index = getIndexKey(key);
+        if (table[index] != null) {
+            Node<K, V> current = table[index];
+            if (Objects.equals(current.key, key)) {
+                current.value = value;
+                return;
+            }
+            while (current.next != null) {
+                current = current.next;
+                if (Objects.equals(current.key, key)) {
+                    current.value = value;
+                    return;
```

Comment:
You do the same action twice, when you check the first key and then the others in a loop.